Apps/Src/ReadTritium.c

```diff
@@ -127,3 +121,52 @@ float Motor_RPM_Get(){ //getter function for motor RPM
 float Motor_Velocity_Get(){ //getter function for motor velocity
 	return Motor_Velocity;
 }
+
+
+/**
+ * Error handler functions
+ * Passed as callback functions to the main assertTaskError function by assertTritiumError
+*/
+
+/**
+ * @brief A callback function to be run by the main assertTaskError function for hall sensor errors
+ * restart the motor if the number of hall errors is still less than the RESTART_THRESHOLD.
+ */ 
+static inline void handler_Tritium_HallError(void) {
+	MotorController_Restart(); 
+}
+
+
+/**
+ * @brief   Assert a Tritium error by checking Motor_FaultBitmap
+ * and asserting the error with its handler callback if one exists.
+ *  Can result in restarting the motor (while < RESTART_THRESHOLD number of hall sensor errors)
+ * or locking the scheduler and entering a nonrecoverable fault (all other cases)
+ * @param   motor_err Bitmap with motor error codes to check
+ */
+static void assertTritiumError(tritium_error_code_t motor_err){   
+	static uint8_t hall_fault_cnt = 0; //trip counter, doesn't ever reset
+
+	Error_ReadTritium = motor_err; // Store error code for inspection info
+	if(motor_err == T_NONE) return; // No error, return
+	
+	if(motor_err != T_HALL_SENSOR_ERR){
+		// Assert a nonrecoverable error with no callback function- nonrecoverable will kill the motor and infinite loop
+		assertTaskError(OS_READ_TRITIUM_LOC, motor_err, NULL, OPT_LOCK_SCHED, OPT_NONRECOV);
+		return;
+	}
+
+	hall_fault_cnt++; 
+
+	//try to restart the motor a few times and then fail out
+	if(hall_fault_cnt > RESTART_THRESHOLD){  
```

Comment:
```suggestion
	//try to restart the motor a few times and then fail out
	if(++hall_fault_cnt > RESTART_THRESHOLD){  
```
just a neat little suggestion, not necessary just thought it looked cool